tsl/src/continuous_aggs/materialize.c

```diff
@@ -459,7 +468,31 @@ get_materialization_end_point_for_table(int32 raw_hypertable_id, int32 materiali
 		*materializing_new_range = false;
 		return old_completed_threshold;
 	}
-
+	if (max_time < end_time)
+	{
+		/* limit the end_time to max value from the hypertable's data, adjusted
+		 * by time_bucket width parameter
+		 */
+		Assert(bucket_width > 0);
+		/* overflow check */
```

Comment:
This look a lot better:

I think this could be simplified further bv using `int64_saturating_add` instead of overflow checks:
```
int64 maxtime_corrected =
				ts_time_bucket_by_type(bucket_width, max_time + bucket_width, time_column_type);
if (maxtime_corrected < max_time)
{
    maxtime_corrected = int64_saturating_add(maxtime_corrected, bucket_width)
}
if (maxtime_corrected < end_time)
{
		end_time = maxtime_corrected;
}
```

Reply:
modified.